### PR TITLE
Release/v0.37.8

### PR DESCRIPTION
**Pull request by hvanz**
[CHANGELOG](https://github.com/cometbft/cometbft/blob/247cb21fab2c5668e2b25334066e5e337f31899b/CHANGELOG.md)

Nightly test passed: https://github.com/cometbft/cometbft/actions/runs/9746501179

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments
- [ ] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,25 @@
...
+*July 1, 2024*
+
+This release reverts the API-breaking change to the `Mempool` interface introduced in the last patch
+release (v0.37.7) while still keeping the performance improvement added to the mempool. It also
+includes a minor fix to the RPC endpoints `/tx` and `/tx_search`.
+
+### BREAKING CHANGES
+
+- `[mempool]` Revert adding the method `PreUpdate()` to the `Mempool` interface, recently introduced
+  in the previous patch release. Its logic is now moved into the `Lock` method. With this change,
```
Comment (adizere) [inline CHANGELOG.md:14]: Let's give the specific version numbers. I am not sure I know myself, please double check my suggestion

```suggestion
  in the previous patch release (v0.37.7). Its logic is now moved into the `Lock` method. With this change,
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,25 @@
...
+  the `Mempool` interface is the same as before the previous patch.
```
Comment (adizere) [inline CHANGELOG.md:15]:
```suggestion
  the `Mempool` interface is the same as before the previous patch (i.e., the same as in v0.37.6).
```

**Review by adizere — APPROVED** [review]
tip-top!

**Review by sergio-mena — APPROVED** [review]
🚀

**Pull request merged**